src/compiler/compile/Component.ts

```diff
@@ -143,8 +144,16 @@ export default class Component {
 					.replace(process.cwd(), '')
 					.replace(regex_leading_directory_separator, '')
 				: compile_options.filename);
-		this.locate = getLocator(this.source, { offsetLine: 1 });
-
+		const locator = getLocator(this.source, { offsetLine: 1 });
+		const tracer = compile_options.sourcemap ? new TraceMap(compile_options.sourcemap) : undefined;
+		this.locate = (c) => {
+			let location:{line:number,column:number} = locator(c);
+			if (tracer) {
+				location = originalPositionFor(tracer, location);
+			}
+			location.line += 1; // make line one based
```

Comment:
is this really just zero based vs one based or another cause?

Reply:
I'm wondering why this all still works the same, since you're always adding +1 to the line, regardless of a source map existing or not.

Reply:
you're already adding 1 above: https://github.com/sveltejs/svelte/pull/8362/files#r1135771594

so I think this is really adding 2 which seems wrong to me

Reply:
in a locally linked vite project with sveltekit-ts it produces correct positions in __svelte_meta.

I agree that this looks off and so far havn't been able to understand why it produces the correct result.

Reply:
perhaps we need to remove this `- 1` instead of adding an additional `+ 1`?

https://github.com/sveltejs/svelte/blob/22eaaf2c691858d1d5879bba0225261633a78e68/src/compiler/compile/Component.ts#L512